Meta/Lagom/Fuzzers/FuzzTar.cpp

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2022, Idan Horowitz <idan.horowitz@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/MemoryStream.h>
+#include <LibArchive/TarStream.h>
+#include <stdio.h>
+
+extern "C" int LLVMFuzzerTestOneInput(uint8_t const* data, size_t size)
+{
+    InputMemoryStream input_stream(ReadonlyBytes { data, size });
+    Archive::TarInputStream tar_stream(input_stream);
+
+    if (!tar_stream.valid())
+        return 0;
+
+    for (; !tar_stream.finished(); tar_stream.advance()) {
+        auto const& header = tar_stream.header();
+
+        if (!header.content_is_like_extended_header())
+            continue;
+
+        switch (header.type_flag()) {
+        case Archive::TarFileType::GlobalExtendedHeader: {
+            auto result = tar_stream.for_each_extended_header([&](StringView, StringView) {});
+            if (result.is_error())
+                return 0;
+            break;
+        }
+        case Archive::TarFileType::ExtendedHeader: {
+            auto result = tar_stream.for_each_extended_header([&](StringView, StringView) {});
+            if (result.is_error())
+                return 0;
+            break;
+        }
```

Comment:
Can we combine those?

```suggestion
        case Archive::TarFileType::ExtendedHeader:
        case Archive::TarFileType::GlobalExtendedHeader: {
            auto result = tar_stream.for_each_extended_header([&](StringView, StringView) {});
            if (result.is_error())
                return 0;
            break;
        }
```

Reply:
Very good point, not sure how I missed that one :sweat_smile:

Reply:
Alternatively, since we just checked for `!is_like_extended_header` earlier, that means that we definitely are like an extended header now, so `for_each_extended_header` has to work or something else is wrong entirely.

Reply:
And, if we want to make the connection a bit clearer, we could just do something like the following:

```
        [...]

        if (header.content_is_like_extended_header()) {
            auto result = tar_stream.for_each_extended_header([&](StringView, StringView) {});
            if (result.is_error())
                return 0;
            continue;
        }
    }
```